python/Makefile

```diff
@@ -18,6 +19,11 @@ develop: setup ## Install Python binding of delta-rs
 	$(info --- Maturin develop ---)
 	maturin develop
 
+.PHONY: setup-dev
+setup-dev: setup
+	maturin build
+	python -m pip install $(TARGET_WHL)[devel,pandas]
+
```

Comment:
Since the `maturin develop` command does not work because the optional dependencies are required (optional dependencies are required... 😄) to set up our local development environment, I would suggest replacing the [develop](https://github.com/delta-io/delta-rs/blob/83aeabbaed788848c2e8091de6cc4182518b91ec/python/Makefile#L17) which is useless by something like this:

```
.PHONY: develop
develop: setup ## Install Python binding of delta-rs
	$(info --- Maturin build ---)
	maturin build
	pip install $(TARGET_WHEEL)'[devel, pandas]'

```

Reply:
I was worried about replacing an existing command, but it makes sense to me 😄